galcheat/zeropoints.py

```diff
@@ -0,0 +1,55 @@
+import math
+
+from astropy import units as u
+from speclite.filters import ab_reference_flux, load_filter
+
+from galcheat.helpers import available_surveys, get_survey
+
+speclite_available_surveys = ["DES", "Euclid_VIS", "HSC", "Rubin"]
+speclite_survey_prefixes = ["decam2014-", "Euclid-", "hsc2017-", "lsst2016-"]
+
+
+def calculate_zero_point(band_name, B0=24):
+    """Compute the zeropoint of a given filter with speclite"""
+    filt = load_filter(band_name)
+    return (filt.convolve_with_function(ab_reference_flux) * 10 ** (-0.4 * B0)).to(
+        1 / (u.s * u.m**2)
+    )
+
+
+def check_zeropoints():
+    """Recompute the zeropoints with speclite and the effective area
+    and compare them to their current values
+    """
+
+    for survey_name in available_surveys:
+        if survey_name in speclite_available_surveys:
+            survey = get_survey(survey_name)
+            print(survey_name, ":")
+
+            speclite_prefix = speclite_survey_prefixes[
+                speclite_available_surveys.index(survey_name)
+            ]
+
+            survey_filters = survey.get_filters()
+            for filt_name in survey_filters:
+                speclite_filt_name = f"{speclite_prefix}{filt_name}"
+                zp_24 = (
+                    calculate_zero_point(speclite_filt_name)
+                    * survey.effective_area
+                    * 1
```

Comment:
Is this the 1 e- ? There is a unit for that (just in case).

Reply:
So here I am not totally sure neither. The `calculate_zero_point` function outputs the zero point in 1/(s.m^2). Multiplying by the effective area gives the zero point in 1/s, which I interpret as a throughput per second. Then I multiply by 1s to obtain for the duration of 1s, convert it to our magnitude convention and set it to a magnitude unit. Does it make sense ?

Reply:
Committed a slight change to put the second next to the one to avoid confusion.